core/solver/bicgstab.cpp

```diff
@@ -140,16 +142,18 @@ void Bicgstab<ValueType>::apply_impl(const LinOp *b, LinOp *x) const
         ++iter;
         this->template log<log::Logger::iteration_complete>(this, iter, r.get(),
                                                             dense_x);
+        rr->compute_dot(r.get(), rho.get());
+        rho_abs = rho->compute_absolute();
```

Comment:
Can we take care of computing the absolute in-place in the stopping criteria? This would eliminate the need for additional alloctions for the absolute vector.

Reply:
Good point, would be nice to have that in the stopping criterion.

Reply:
Good catch. I have updated this now.